### ISSUE TITLE
Deleting any page with children is broken

### ISSUE THREAD

**Issue by sjclark** [statamic/cms#6612]
### Bug description

Attempting to **delete any page that has any number of children will result in the error** "Root page cannot have children"
<img width="399" alt="Screen Shot 2022-09-01 at 5 37 24 pm" src="https://user-images.githubusercontent.com/4531017/187858821-4fff548e-a7a8-4188-bd41-d58b56ffbf31.png">

This bug occurs regardless of checking the option to delete any child pages;
<img width="344" alt="Screen Shot 2022-09-01 at 5 36 25 pm" src="https://user-images.githubusercontent.com/4531017/187858828-b1543de0-8490-4f69-a9f0-350c05c0053e.png">

This occurs even on a completely fresh Statamic install with no addons - and seems to occur across a wide range of revisions (every site I currently had handy).

https://user-images.githubusercontent.com/4531017/187860475-7d1505d9-5a25-4288-b354-b03760cbcc30.mov

### How to reproduce

1. Add any page
2. Add any child page
3. Delete the parent
4. When trying to save changes receive error `"Root page cannot have children"`

### Environment

```yaml
Application Name: Statamic
Laravel Version: 9.27.0
PHP Version: 8.1.9
Composer Version: 2.2.9
Environment: local
Debug Mode: ENABLED
URL: mysite.test
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

### Additional details

In **Tree** view this error is limited to clicking "Save changes"
In **List** view this error occurs directly after clicking "Delete"

Additionally, it appears something is going wrong with the `"Delete Child Entry"` option, as in **Tree** view deleting the child page and then the parent, and **then** clicking save, works correctly.

https://user-images.githubusercontent.com/4531017/187860869-82698801-f9e3-404d-a32b-bb5f48a5027c.mov

**Comment by jelleroorda** [comment]
Related to #4453, perhaps duplicate

**Comment by sjclark** [comment]
> Related to #4453, perhaps duplicate

Bah! No idea how I missed that when searching, yeah, guess it’s a duplicate, amazed it’s been broken that long!

**Comment by edalzell** [comment]
Closed in favour of #4453
